ec2/ec2.go

```diff
@@ -127,8 +127,46 @@ type xmlErrors struct {
 
 var timeNow = time.Now
 
-// resp = response structure that will get inflated by XML unmarshaling.
+// query performs a query with backoff if a rate limit error is received.
 func (ec2 *EC2) query(params map[string]string, resp interface{}) error {
+	const maxAttempts = 30
```

Comment:
30 is too much, total time for the call before it errors is 25minutes.

Reply:
10 would be sufficient.

Reply:
Yeah, we have this in the ec2 provider code:

```
var deleteSecurityGroupInsistently = func(inst SecurityGroupCleaner, ctx context.ProviderCallContext, group ec2.SecurityGroup, clock clock.Clock) error {
	err := retry.Call(retry.CallArgs{
		Attempts:    30,
		Delay:       time.Second,
		MaxDelay:    time.Minute, // because 2**29 seconds is beyond reasonable
		BackoffFunc: retry.DoubleDelay,
...
```
I was being consistent with what we already do for retry there, but 30 does seem too much.

Reply:
Changed to 10